docs/root/configuration/listeners/listener_filters/proxy_protocol.rst

```diff
@@ -29,10 +33,35 @@ If there is a protocol error or an unsupported address family
 Statistics
 ----------
 
-This filter emits the following statistics:
+This filter emits the following general statistics, rooted at *downstream_cx_proxy_proto*
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 4, 1, 8
+
+  not_found_disallowed, Counter, "Total number of connections that don't contain the PROXY protocol header and are rejected."
+  not_found_allowed, Counter, "Total number of connections that don't contain the PROXY protocol header, but are allowed due to :ref:`allow_requests_without_proxy_protocol <envoy_v3_api_field_extensions.filters.listener.proxy_protocol.v3.ProxyProtocol.allow_requests_without_proxy_protocol>`."
+
+
+The filter also emits the statistics rooted at *downstream_cx_proxy_proto.versions.<version>*
+for each matched PROXY protocol version. Proxy protocol versions include ``v1`` and ``v2``.
 
 .. csv-table::
   :header: Name, Type, Description
-  :widths: 1, 1, 2
+  :widths: 4, 1, 8
+
+  found, Counter, "Total number of connections where the PROXY protocol header was found and parsed correctly."
+  disallowed, Counter, "Total number of ``found`` connections that are rejected due to :ref:`disallowed_versions <envoy_v3_api_field_extensions.filters.listener.proxy_protocol.v3.ProxyProtocol.disallowed_versions>`."
+  error, Counter, "Total number of connections where the PROXY protocol header was malformed (and the connection was rejected)."
+
+The filter also emits the following legacy statistics, rooted at its own scope:
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 4, 1, 8
+
+  downstream_cx_proxy_proto_error, Counter, "Total number of connections with proxy protocol errors, i.e. ``v1.error``, ``v2.error``, and ``not_found_disallowed``."
 
-  downstream_cx_proxy_proto_error, Counter, Total proxy protocol errors
+.. attention::
+  The legacy statistics are deprecated and kept for backwards compatibility.
```

Comment:
Please remove this sentence. It's not deprecated; deprecated stuff has a plan for removal, and I think we'll just keep this forever since it's relatively easy to do so. It's just rooted in the wrong place.

Reply:
Done